pkg/k8shandler/common.go

```diff
@@ -26,7 +26,10 @@ func addOwnerRefToObject(o metav1.Object, r metav1.OwnerReference) {
 func getImage(commonImage string) string {
 	image := commonImage
 	if image == "" {
-		image = utils.LookupEnvWithDefault("ELASTICSEARCH_IMAGE", elasticsearchDefaultImage)
+		image = utils.LookupEnvWithDefault("ELASTICSEARCH_IMAGE", "")
+		if image == "" {
```

Comment:
is it really necessary to change this to panic now instead of falling back to a default? the majority of the time it will be on the `elasticsearch` CR

Reply:
> is it really necessary to change this to panic now instead of falling back to a default? the majority of the time it will be on the `elasticsearch` CR

Is there a reason that we should keep the default image in two places, code and env var?  The expectation is the operator will be deployed knowing what default image it supports.  Is this different then the requirement to provide 'WATCH_NAMESPACE' or 'POD_NAME'?  This code should likely be pushed into a start routine where we panic early instead of waiting to process a CR.

Reply:
to me we shouldn't fail. we want to lockstep the operator version and the ES image version, so I feel like we would fall back on what we say we support rather than fail if an ENV VAR isn't provided

Reply:
> to me we shouldn't fail. we want to lockstep the operator version and the ES image version, so I feel like we would fall back on what we say we support rather than fail if an ENV VAR isn't provided

We do this with the manifest

Reply:
I'd be inclined to keep the default in code. If there is a sane default that is known at coding time, I prefer to bake it in so the program does something sane even with no external config. You don't want to duplicate the default in 2 places - so leave the env var unset for default behaviour. That way the default can be changed in one place even if there are many CRDs using the default behavior.

Reply:
The difference here is the "default" baked into the code is never really known, its a generic image tag that is the "upstream" source.  Contrary, the version pegged to the image is always provided by the release process which substitutes the value checked in the manifest to a value that is built during the release.  This means the value provided by an env var from the manifest the only value that is ever a released image.